src/NuGet.Protocol.Core.v3/HttpHandlerResourceV3Provider.cs

```diff
@@ -66,12 +64,7 @@ private HttpClientHandler TryGetCredentialAndProxy(PackageSource packageSource)
                 && !String.IsNullOrEmpty(packageSource.UserName)
                 && !String.IsNullOrEmpty(packageSource.Password))
             {
-                var cache = new CredentialCache();
```

Comment:
@yishaigalatzer Right, with the current implementation, you can't host two different feeds that require different credentials on the same host. We need to redesign our authentication implementation to support this scenario.